torch/nn/_functions/bilinear.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from torch.autograd import Function
+
+
+class Bilinear(Function):
+
+    def forward(self, input1, input2, weight, bias=None):
+        self.save_for_backward(input1, input2, weight, bias)
+
+        output = input1.new()
```

Comment:
in the new nn (i.e. not legacy nn), we do not need to use temporary buffers.
Also, torch.mm resizes it's output, so buf doesn't need to be pre-resized
So this function can be reduced to:

```
self.save_for_backward(input1, input2, weight, bias)

output = input1.new(input1.size(0), weight.size(0))
buf = input1.new()

# compute output scores:
for k, w in enumerate(weight):
    torch.mm(input1, w, out=buf)
    buf.mul_(input2)
    torch.sum(buf, 1, out=output.narrow(1, k, 1))
if bias is not None:
    output.add_(bias.expand_as(output))
return output
```